modules/subschema/src/main/scala/com.snowplowanalytics/iglu.schemaddl/jsonschema/subschema/package.scala

```diff
@@ -173,7 +174,7 @@ package object subschema {
         case (Some(m1), Some(m2)) => s".{${m1.value},${m2.value}}"
         case (None, Some(m2))     => s".{0,${m2.value}}"
         case (Some(m1), None)     => s".{${m1.value},}"
-        case (None, None)         => s".{0}"
+        case (None, None)         => s".{0,}"
```

Comment:
I can confirm this change breaks a test on the msc-backend https://github.com/snowplow-product/msc-backend/blob/114a9ae646db97ff5e2c0132ca33af8fd9d0e53f/src/test/scala/com/snowplowanalytics/msc/modules/trackingscenarios/validation/CompatibilityCheckingSpec.scala#L276

Reply:
Hm, yes, something is wrong with this, test seems to be correct

Reply:
I think I've just gone full circle with this one. Back to a time before we looked at it @patofp. There is something wrong here, but I cant put my finger on it.

using `.{0}` makes `{type:string, enum:['a']}` a valid subset of `{type:string, maxLength:2}` which is the case we fixed for.

but makes `{type:string, maxLength:1, minLength:1}` not a valid subset of `{type:string}`
and other combinations of max and min length get a bit wonky,

`{type:string, maxLength:1}` not compatible with `{type:string}` when it should be
also
`{type:string}` compatible with `{type:string, maxLength:1}` which isnt right

Reply:
I re-read the rules around min and max length strings. It looks like I was constructing the comparison wrong, missing a `∩`.

Could you have another check @patofp ? Also my string test cases need a careful look from someone else, I think I'm going a little column blind.

Reply:
Yeah, I'm on it, I'm trying to debug the previous cases. 😃

Reply:
Let's take the case `{ "type": "string" }`, this schema is equivalent to `{ "type": "string", "minLength": 0, "pattern": ".*" }`. By applying this rule:
<img width="1038" alt="Screenshot 2023-04-12 at 12 57 27" src="https://user-images.githubusercontent.com/6108743/231438020-3f4e26d4-0a53-4e0b-a0bb-9265adfe8940.png">

We would need to calculate the intersection between `'.*' intersect '.{0}' =  '.{0}'`. But that doesn't make much sense as the only inhabitant of that space is the empty string. 🤔 

I think this rule makes sense for a > 0. My guess is we should unfold that rule into two, one for a > 0 and one for a = 0:

`s.type = string, s.pattern = p, s.minLength = 0, maxLength !E dom(s) => s -> {type: string, pattern: p}`

Reply:
I think you are right @jamessnowplow, it can't be `.{a}` but `.{a,}`. 
- `{"type": "string"} => {"type": "string", "pattern": ".{0,}"}`
- `{"type:": "string", "minLength": 2 } => { "type": "string" "pattern": ".{2,}"}`
- `{"type": "string", "maxLength": 2 } => {"type": "string", "pattern": ".{0,2}"}`
- `{"type": "string", "minLength": 2, "maxLength": 3 } => {"type": "string", "pattern": ".{2,3}"}`